AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/MainActivity.java

```diff
@@ -0,0 +1,74 @@
+package solutions.alterego.androidbound.example.support;
+
+import com.alterego.advancedandroidlogger.interfaces.IAndroidLogger;
+
+import android.graphics.Typeface;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+
+import solutions.alterego.androidbound.android.AndroidViewModel;
+import solutions.alterego.androidbound.example.support.imageloader.UILImageLoader;
+import solutions.alterego.androidbound.example.support.util.AdvancedAndroidLoggerAdapter;
+import solutions.alterego.androidbound.example.support.viewmodels.MainActivityViewModel;
+import solutions.alterego.androidbound.interfaces.ILogger;
+import solutions.alterego.androidbound.interfaces.IViewBinder;
+
+
+public class MainActivity extends AppCompatActivity {
+
+    public static final String LOGGING_TAG = "TEST_APP";
+
+    private static final IAndroidLogger.LoggingLevel LOGGING_LEVEL = IAndroidLogger.LoggingLevel.VERBOSE;
+
+    private AndroidViewModel mViewModel;
+
+    private IViewBinder mViewBinder;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ILogger logger = new AdvancedAndroidLoggerAdapter(LOGGING_TAG, LOGGING_LEVEL);
+
+        mViewBinder = ExampleApplication.getViewBinder();
+
+        mViewBinder.getFontManager().setDefaultFont(Typeface.createFromAsset(getAssets(), "Roboto-Bold.ttf"));
+        mViewBinder.getFontManager().registerFont("light", Typeface.createFromAsset(getAssets(), "Roboto-Light.ttf"));
+        mViewBinder.getFontManager().registerFont("italic", Typeface.createFromAsset(getAssets(), "Roboto-Italic.ttf"));
+        mViewBinder.getFontManager().registerFont("bold", Typeface.createFromAsset(getAssets(), "Roboto-Bold.ttf"));
+        mViewBinder.setImageLoader(new UILImageLoader(this, null));
+
+        mViewModel = new MainActivityViewModel(this, logger);
+        mViewModel.onCreate(savedInstanceState);
+
+        View view = mViewBinder.inflate(this, mViewModel, R.layout.activity_main, null);
+        setContentView(view);
+
+        setTitle("MainActivity");
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        mViewModel.onGotFocus();
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        mViewModel.onLostFocus();
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        mViewModel.onSaveInstanceState(outState);
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        mViewModel.dispose();
+        mViewBinder.disposeOf(this);
```

Comment:
why do you need this line?

Reply:
it makes sure all the references are released. most of them shouldn't be a problem, but I don't know about the `FontManager` that has typefaces, so this is a "better safe than sorry" thing.